Lib/test/test_email/test_policy.py

```diff
@@ -245,6 +245,21 @@ def test_non_ascii_chars_do_not_cause_inf_loop(self):
             'Subject: \n' +
             12 * ' =?utf-8?q?=C4=85?=\n')
 
+    def test_short_maxlen_error(self):
+        # RFC 2047 chrome takes up 7 characters, plus the length of the charset
+        # name, so folding should fail if maxlen is lower than the minimum
+        # required length for a line
+
+        # Note: This is only triggered when there is a single word longer than
+        # maxlen, hence the 1234567890 at the end of this whimsical subject
```

Comment:
Slight change to the comment to make it a bit more clear why this bug is trigerred when there is a single world longer than `maxlen`.

```
# Note: This is only triggered when there is a single word longer than
# max_line_length. This is because when we encounter a word longer than max_line_length,
# it is broken down into encoded words to fit max_line_length. If the max_line_length isn't
# large enough to even  contain RFC 2047 chrome (`?=<charset>?q??=`), we fail.
# Hence the 1234567890 at the end of this whimsical subject.
```

Just to make it clear that Header can have words longer than max line length, as  long as `max_line_length > 7 + len(charset) + 1`.

Reply:
(There's an extra space in your comment: "even  contain"

Reply:
`HeaderParseError` seems right to me too, but maybe @bitdancer has a different opinion?

Reply:
Actually, the phrase "even  contain RFC 2047 chrome (`?=<charset>?q??=`)" should probably be "even contain RFC 2047 chrome plus one character(`?=<charset>?q?x?=`)"